pkg/controller/controller.go

```diff
@@ -22,33 +22,42 @@ import (
 
 	"github.com/golang/glog"
 
+	"istio.io/broker/pkg/model/config"
 	"istio.io/broker/pkg/model/osb"
 )
 
-const (
-	demoCatalogFilePath = "example"
-	catalogFileName     = "demo_catalog.json"
-)
-
 // Controller data
 type Controller struct {
+	config.BrokerConfigStore
 }
 
 // CreateController creates a new controller instance.
-func CreateController() (*Controller, error) {
-	return new(Controller), nil
+func CreateController(config config.BrokerConfigStore) (*Controller, error) {
+	return &Controller{config}, nil
 }
 
 // Catalog serves catalog request and generate response.
 func (c *Controller) Catalog(w http.ResponseWriter, _ *http.Request) {
-	glog.Infof("Get Service Broker Catalog...")
-	var catalog osb.Catalog
+	glog.Infof("Fetching Service Broker Catalog...")
+	cat := c.catalog()
+	glog.V(2).Infof("Got catalog\n %#v", cat)
+	writeResponse(w, http.StatusOK, cat)
```

Comment:
Should `writeResponse` set the `content-type` to `application/json`?

Reply:
done